components/builder-core/src/api.rs

```diff
@@ -380,11 +401,13 @@ fn do_group_promotion(
 
     let channel = match conn.route::<OriginChannelGet, OriginChannel>(&ocg) {
         Ok(c) => {
-            if !check_origin_access(session_id, origin).map_err(
-                Error::NetError,
-            )?
-            {
-                return Err(Error::OriginAccessDenied);
+            if let Some(session_id) = session_id_opt {
+                if !check_origin_access(session_id, origin).map_err(
+                    Error::NetError,
+                )?
+                {
+                    return Err(Error::OriginAccessDenied);
+                }
```

Comment:
If I'm following this correctly, if no session_id is passed in, then we skip the check to make sure the thing making this request has permission to access the origin in question.  Is that because the thing making the request will be another builder service?  My impression reading this code is that we want to perform this check when there is a user making this request, and thus, has a session id, but builder services don't have a session and should be implicitly trusted, which is why it's safe to skip this check here.  Is that assumption correct or have I misread this?

Reply:
Yes, that's exactly right and that is the intent.